src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java

```diff
@@ -41,7 +41,19 @@ protected UUID _deserialize(String id, DeserializationContext ctxt) throws IOExc
              *   length it must have...
              */
             if (id.length() == 24) {
-                byte[] stuff = Base64Variants.getDefaultVariant().decode(id);
+                byte[] stuff = Base64Variants.getDefaultVariant().decode(id
+                    //cater for url safe chars
+                    .replace("-", "+")
```

Comment:
Isn't this remarkably inefficient way to do this, using dynamically compiled regexp replaced and constructing Strings on the fly? Should be easy enough to change to character-based alternative.

But overall I am not sure how safe it is to start replacing characters here. I'll have to think about this a bit.

Reply:
`replace(String, String)` is actually not regex, and it even delegates to `replace(char, char)` if both strings are length 1. still should move to the char version though.

Reply:
Ah ok. I misremembered that then. Should probably move logic to separate helper method regardless.

Reply:
Ok, changed to chars for the replacement now + separated the logic to helper methods.